### PR TITLE
6 post/api/articles/article id/comments

### PR DESCRIPTION
**Pull request by LukeRiches**

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+        expect(typeof comment.article_id).toBe("number")
+
+        expect(comment).toHaveProperty("author")
+        expect(typeof comment.author).toBe("string")
+
+        expect(comment).toHaveProperty("votes")
+        expect(typeof comment.votes).toBe("number")
+
+        expect(comment).toHaveProperty("created_at")
+        expect(typeof comment.created_at).toBe("string")
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: This test is a good foundation but could be improved in a few ways:
- If you're testing `typeof comment.property...` you don't also need to test `.toHaveProperty`, if it doesn't have the property then the typeof test will fail so you only need that one for each property
- In the case of some of these properties, we should know exactly what they are. For example, the _article_id_ should be the one from the path, `1`. The author should be the one from the comment object, `lurker`. Replacing the `typeof` tests for more specific tests on those properties will make your test more secure

File: __tests__/app.test.js [hunk 2]
```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+    const badComment = {
+      username : "Test",
+      body : "body..."
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(badComment)
+      .expect(404)
+      .then((response) => {
+        expect(response.body.msg).toBe('User does not exist');
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Brilliant

File: __tests__/app.test.js [hunk 3]
```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+    };
+    return request(app)
+      .post('/api/articles/1/comments')
+      .send(badComment)
+      .expect(400)
+      .then((response) => {
+        expect(response.body.msg).toBe('Invalid request body');
+      });
+  });
+});
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Really good testing, another you should add on is a test for when the _article_id_ is not a valid data type, for example `/api/articles/banana/comments`

File: __tests__/app.test.js [hunk 4]
```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+
+describe('GET /api/users', () => {
+  test('Should respond with a users array of users objects ', () => {
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Testing a users endpoint is superfluous to the ticket. I will explain in the next few comments why you don't need to make a query to the users table to complete the error testing on this endpoint.
Please copy the code for this to another branch and then remove it from this one

File: __tests__/app.test.js [hunk 5]
```diff
@@ -206,6 +211,138 @@ describe('GET /api/articles/:article_id/comments', () => {
...
+      });
+  });
+  /**  Once queries are added further testing for: 
+  - When query exists but not a valid input
+  - When given a non existent query sends an appropriate error and error message
+  */ 
+});
+
+describe('GET /api/users/:username', () => {
+  test('should respond with a single user object', () => {
```
Comment (Pipe-Smoking-Rabbit) [inline __tests__/app.test.js]: Same goes for these tests, copy and remove ❎

File: Controllers/comments-controller.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+}
+
+function postComment(req, res, next){
+    const newComment = req.body;
+    const username = newComment.username;
+    const {article_id} = req.params;
+
+    function bodyCheck(newComment){
+        
+        if (!newComment.hasOwnProperty("username") ||!newComment.hasOwnProperty("body") || Object.keys(newComment).length !== 2) {
```
Comment (Pipe-Smoking-Rabbit) [inline Controllers/comments-controller.js]: This `if` statement and the `Promise.all` below are not necessary for this controller. In the seed function, the comments table has _author_ & _article_Id_ as _foreign keys_ that reference the `users` and `articles` tables respectively. 
As such, all you need for the query on this endpoint is the `INSERT INTO...`, if anything is missing or cannot be found in the database, PSQL will throw an error to the catch with a code of `23503` or `23502`

**Review by Pipe-Smoking-Rabbit — CHANGES_REQUESTED** [review]
You've worked really hard on this ticket and done good testing. Unfortunately some of what you have is not necessary for the ticket and can be removed. 
Please submit another PR so we can review your changes once they've been made 👍

**Review by Beth-McCormack — APPROVED** [review]

**Pull request merged**